src/components/time-picker/time-picker.tsx

```diff
@@ -0,0 +1,382 @@
+import * as keycode from 'keycode';
+import * as React from 'react';
+import {findDOMNode} from 'react-dom';
+import {SBComponent} from 'stylable-react-component';
+import {Stepper} from '../stepper';
+import styles from './time-picker.st.css';
+import {
+    Ampm, Format, isTimeSegment,
+    isValidValue, pad2, Segment, selectionIndexes, TimeSegment, to24, toAmpm
+} from './utils';
+
+export interface Props {
+    value?: string;
+    onChange?: (value: string) => void;
+    format?: Format;
+    placeholder?: string;
+    disabled?: boolean;
+    label?: string;
+    name?: string;
+}
+
+export interface State {
+    inputValue: string;
+    focus: boolean;
+    hh?: number;
+    mm?: number;
+    ampm: Ampm;
+    currentSegment: Segment;
+}
+
+const isTouch = typeof window === 'object' && 'orientation' in window;
+const is12TimeFormat = /AM|PM/.test(new Date().toLocaleTimeString());
+const ampmLabels = {
+    [Ampm.AM]: 'AM',
+    [Ampm.PM]: 'PM',
+    [Ampm.NONE]: ''
+};
+
+function segmentsToInputValue({hh, mm, ampm}: {hh?: number, mm?: number, ampm: Ampm}) {
+    if (hh === undefined && mm === undefined) {
+        return '';
+    }
+    return [
+        hh ? pad2(hh) : '00',
+        ':',
+        mm ? pad2(mm) : '00',
+        ' ',
+        ampmLabels[ampm]
+    ].join('').trim();
+}
+
+function defaultValue(format: Format, hh: string = '00') {
+    return hh + ':00' + (format === 'ampm' ? ' AM' : '');
+}
+
+function propsValueToSegments(value?: string, format?: Format) {
+    const isAmpm = format === 'ampm';
+    if (!value) {
+        return {
+            ampm: isAmpm ? Ampm.AM : Ampm.NONE
+        };
+    }
+    const [hh24, mm] = value.split(':').map(Number);
+    const {hh, ampm} = toAmpm(hh24);
+    return {
+        hh: isAmpm ? hh : hh24,
+        ampm: isAmpm ? ampm : Ampm.NONE,
+        mm
+    };
+}
+
+@SBComponent(styles)
+export default class TimePicker extends React.Component<Props, State> {
+    public static defaultProps = {
+        format: is12TimeFormat ? 'ampm' : '24hr',
+        disabled: false
+    };
+    private input: HTMLInputElement | null;
+    private lastValue: string | undefined;
+
+    constructor(props: Props) {
+        super();
+        this.lastValue = props.value;
+        this.state = {
+            focus: false,
+            currentSegment: 'hh',
+            ...this.getInitialState(props)
```

Comment:
not the best name for this function, but it return only `hh, mm, ampm, inputValue`